wpengine/faustjs
# Core Code Block

## Tasks


## Description



* Provides reference implementation of CoreCode block. 
* Uses `getStyles` helper
* Adds Unit tests.

## Related Issue(s):



## Testing

[You can reference this commit that has all the steps from testing below.](https://github.com/wpengine/faustjs/pull/1441/commits/627d743f4a1545f58e6d7d3637fae5b261ce698c)


* Follow instructions to configure the `blocks` package in the example project. https://faustjs.org/docs/gutenberg/getting-started
* Generate the global stylessheet and import it into the project. `faust generateGlobalStylesheet`.
* Import your `theme.json` from your WordPress template
```_app.js
import { WordPressBlocksProvider, fromThemeJson } from '@faustwp/blocks';
import themeJson from '../theme.json';
import '../globalStylesheet.css';

export default function MyApp({ Component, pageProps }) {
  const router = useRouter();
  return (
    <FaustProvider pageProps={pageProps}>
      <WordPressBlocksProvider
        config={{
          blocks,
          theme: fromThemeJson(themeJson),
        }}>
        <Component {...pageProps} key={router.asPath} />
      </WordPressBlocksProvider>
    </FaustProvider>
  );
}
```
* Comment out all the styles in the `examples/next/faustwp-getting-started/styles/_blocks.scss` since they will interfere with the global stylesheet.
* Import the blocks:
```js
import { coreBlocks } from '@faustwp/blocks';

export default {
  ...coreBlocks,
};
```
* Make sure you use the block fragment in the page query:

```graphql
${components.CoreCode.fragments.entry}
...
editorBlocks {
        __typename
        renderedHtml
        id: clientId
        parentClientId
        ...${components.CoreCode.fragments.key}
      }
```
* Use the `WordPressBlocksViewer`:
```
<Container>
            <ContentWrapper>
              <WordPressBlocksViewer blocks={blocks}/>
            </ContentWrapper>
          </Container>
```
* Go to WordPress site and create several code blocks combining styles and colors. 

## Screenshots

Headless site on the left, WordPress on the right. You may notice the initial border and padding isn't being rendered. This is because these are theme related styles (twentytwentytwo), which aren't applied yet. We do have a ticket in the backlog to handle theme related styles. https://wpengine.atlassian.net/browse/MERL-870

<img width="2560" alt="Screenshot 2023-06-01 at 1 57 38 PM" src="https://github.com/wpengine/faustjs/assets/5946219/52ab292b-9c17-4f2c-a5de-643c1b36e45a">



## Documentation Changes



## Dependant PRs
